src/analysis/AnalysisNotificationManager.ts

```diff
@@ -120,30 +141,43 @@ const AppErrorNotification = ({ app }) => {
   ]);
 };
 
-function RuntimeManager({ namespace, name, runtimes, apps }) {
-  const prevRuntimes = usePrevious(runtimes);
-  const prevApps = usePrevious(apps);
+interface AnalysisNotificationManagerProps extends PropsWithChildren {
+  namespace: string;
+  name: string;
+  runtimes: Runtime[];
+  apps: App[];
+}
+
+export const AnalysisNotificationManager = (props: AnalysisNotificationManagerProps) => {
+  const { namespace, name, runtimes, apps } = props;
+  const prevRuntimes: Runtime[] = usePrevious(runtimes) || [];
+  const prevApps: App[] = usePrevious(apps) || [];
 
   useEffect(() => {
-    const runtime = getCurrentRuntime(runtimes) || {};
```

Comment:
This attempt to make runtime not undefined was a bit funny... we defaulted it to an empty object which is not the same as a runtime object. There were a lot of bugs surrounding this that typing it as the return of `getCurrentRuntime` (`Runtime | undefined`) trivially fixed.